### PR TITLE
Add support for multiple backends

### PR DESCRIPTION
**Pull request by emveepee**
Following model from @ksooo and pvr.hts 591 PR https://github.com/kodi-pvr/pvr.hts/pull/591  move to multi instance model.  Opted to use tinyxml2 for the migration.

### CODE REVIEW

File: src/Channels.cpp [hunk 1]
```diff
@@ -16,6 +16,13 @@ using namespace NextPVR::utilities;
 
 /** Channel handling */
 
+Channels::Channels(const std::shared_ptr<InstanceSettings>& settings, Request& request) :
+  m_settings(settings),
+  m_request(request)
+{
+
```
Comment (phunkyfish) [inline src/Channels.cpp]: Remove blank line

File: src/EPG.cpp [hunk 2]
```diff
@@ -19,6 +19,15 @@ using namespace NextPVR::utilities;
 /************************************************************/
 /** EPG handling */
 
+EPG::EPG(const std::shared_ptr<InstanceSettings>& settings, Request& request, Recordings& recordings, Channels& channels) :
+  m_settings(settings),
+  m_request(request),
+  m_recordings(recordings),
+  m_channels(channels)
+{
+
```
Comment (phunkyfish) [inline src/EPG.cpp]: Remove blank line

File: src/Timers.cpp [hunk 3]
```diff
@@ -19,6 +19,15 @@ using namespace NextPVR::utilities;
 /************************************************************/
 /** Timer handling */
 
+Timers::Timers(const std::shared_ptr<InstanceSettings>& settings, Request& request, Channels& channels, cPVRClientNextPVR& pvrclient) :
+  m_settings(settings),
+  m_request(request),
+  m_channels(channels),
+  m_pvrclient(pvrclient)
+{
+
```
Comment (phunkyfish) [inline src/Timers.cpp]: Remove blank line

**Review by phunkyfish — APPROVED** [review]
Just 3 minors, nice work. Now I just have to find the time to add this to my addons!

File: src/Channels.cpp (hunk 1 above)
Comment (emveepee) [reply src/Channels.cpp]: Done

**Comment by emveepee** [comment]
@phunkyfish I added one commit for review.  I also modified the "Instance folders" comit  to use special;//profile not special://userdata

**Review by phunkyfish — APPROVED** [review]

**Pull request merged**